File: 백준/Silver/10845. 큐/큐.py
import sys
from collections import deque

input = sys.stdin.readline

n = int(input())
result = deque()

for i in range(n):
    arg = input().split()
    
    if arg[0] == 'push':
        result.append(arg[1])
        
    elif arg[0] == 'pop':
        if len(result) > 0:
            print(result.popleft())
        else: print(-1)
    
    elif arg[0] == 'size':
        print(len(result))
    
    elif arg[0] == 'empty':
        if len(result) > 0: print(0)
        else: print(1)
        
    elif arg[0] == 'front':
        if len(result) > 0: print(result[0])
        else: print(-1)
    
    elif arg[0] == 'back':
        if len(result) > 0: print(result[len(result)-1])
        else: print(-1)

File: 백준/Silver/10845. 큐/큐.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int n = Integer.parseInt(br.readLine());

        Deque<Integer> queue = new ArrayDeque<>();

        for (int i = 0; i < n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            String command = st.nextToken();

            switch (command) {
                case "push":
                    int num = Integer.parseInt(st.nextToken());
                    queue.offer(num);
                    break;
                case "pop":
                    sb.append(queue.isEmpty() ? -1 : queue.poll()).append("\n");
                    break;
                case "size":
                    sb.append(queue.size()).append("\n");
                    break;
                case "empty":
                    sb.append(queue.isEmpty() ? 1 : 0).append("\n");
                    break;
                case "front":
                    sb.append(queue.isEmpty() ? -1 : queue.peek()).append("\n");
                    break;
                case "back":
                    sb.append(queue.isEmpty() ? -1 : queue.peekLast()).append("\n");
                    break;
            }
        }
        System.out.println(sb);
    }
}
